kube/src/api/dynamic.rs

```diff
@@ -188,6 +188,83 @@ impl TryFrom<DynamicResource> for Resource {
     }
 }
 
+/// Cheaply `Clone`-able string
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+pub enum StringRef {
+    /// Should be used when string is known during compilation
+    Static(&'static str),
+    /// Should be used when string is only known at runtime
+    Dynamic(Arc<str>),
+}
+
+impl StringRef {
+    fn as_str(&self) -> &str {
+        match self {
+            StringRef::Static(s) => *s,
+            StringRef::Dynamic(s) => &*s,
+        }
+    }
+}
+
+/// Represents a type-erased object kind
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+pub struct ErasedKind {
+    /// API group
+    pub group: StringRef,
+    /// Version
+    pub version: StringRef,
+    /// Kind
+    pub kind: StringRef,
+    /// Concatenation of group and version
+    pub api_version: StringRef,
+}
```

Comment:
Maybe we should call this struct what it really is; `GroupVersionKind`.
Sure we have both versions in there, but a constructor should work with just 3 of the 4.

Reply:
Sure, it's a more descriptive name!

I'll rename it and add a constructor.